### PR TITLE
Delayed_start, stall snabb so that peer NIC drivers can fully initialize

### PR DESCRIPTION
**Pull request by petebristow**
An app that delays packet forwarding for a configurable period of time.
Useful when feeding pcaps into a physical nic. The delay lets the peer
NIC the link to come up before sending packets so none are dropped.

Waiting for a linkup in the NicDriver:new() delays the whole process
in every situation and is even worse when a port doesn't link as all the 
timers need to expire.

### CODE REVIEW

**Comment by eugeneia** [comment]
Calling `usleep` will block the whole snabb process as well (timers will not be run for instance). I suggest using [engine.now](http://snabbco.github.io/#engine-core.app) instead:

```
function Delayed_start:new (delay)
   return setmetatable({ start = now()+delay },
                       { __index = Delayed_start})
end

function Delayed_start:push()
   if engine.now() < self.start then return end
   for _ = 1, link.nreadable(self.input.input) do
      link.transmit(self.output.output, link.receive(self.input.input))
   end
end
```

**Comment by petebristow** [comment]
Blocking the whole snabb process with `usleep` was by design. There was an issue floating around a while ago about back pressure was it good / was it bad. I don't remember it reaching a conclusion so it's on my list of things to play with. I wanted to avoid putting back pressure in place with this app or buffering packets in memory, which could go rather wrong if it's being fed by Repeater. 

I'd like to leave it as is if possible, perhaps expanding the comment to make it explicit that it will stall the whole snabb process.

**Comment by eugeneia** [comment]
I don't think blocking is an option as it breaks `engine.main({duration=...})`, but you are right: we need to resolve the back pressure debate! See #656.

As of now anyways, the solution I suggested would work with `Repeater`, as it will only transmit packets when there is room in its output link, e.g. some packets would be buffered in `Delayed_start`.

Cc @lukego

File: src/apps/test/delayed_start.lua [hunk 1]
```diff
@@ -10,14 +10,12 @@ Useful when feeding pcaps into a physical nic. The delay lets the peer
...
+   return setmetatable({ start = engine.now() + delay },
+					  { __index = Delayed_start })
 end
 
 function Delayed_start:push()
-   if self.delay then
-      C.usleep(self.delay)
-      self.delay = nil
-   end
+   if engine.now() < self.start then return end
```
Comment (lukego) [inline src/apps/test/delayed_start.lua]: This looks like a neat solution to me!

**Comment by petebristow** [comment]
@eugeneia Is there anything else you want to see before this can be merged?

**Comment by eugeneia** [comment]
@petebristow I merged it and replaced some tabs with spaces along the way, could you make sure your editor does not emit tabs?

**Pull request merged**